x-pack/plugins/stack_connectors/server/connector_types/jira/service.ts

```diff
@@ -106,6 +106,10 @@ export const createExternalService = (
       fields = { ...fields, parent: { key: incident.parent } };
     }
 
+    if (incident.customFields) {
+      fields = { ...fields, ...incident.customFields };
```

Comment:
This allows users to pass any field they want and even overwrite the previous fields. Although we cannot do much, we can remove from `incident.customFields` any known fields like the `priority` or the `parent`. Wdyt?

Reply:
very good point, i will do that

Reply:
If you have the known fields list, maybe you can validate the customFields against them.
Just removing fields from the payload without notifying the user could be confusing.

Reply:
> If you have the known fields list, maybe you can validate the `customFields` against them.

That is part of the plan for the frontend, we'll validate against known fields from the jira api.

Reply:
The more I think about it the more I realize that, from an API standpoint, these are not custom fields but a way for users to define all the fields we do not support. Something like "Other fields". The reason is that a) users can define any field they want even if it is a custom field or not b) it is difficult for us to distinguish which fields are custom fields and which are not and c) it is difficult to validate against them. What do you all think?

Reply:
@cnasikas I don't see which solution you are advocating for in your comment 😄 

>  The more I think about it the more I realize that, from an API standpoint, these are not custom fields but a way for users to define **all the fields we do not support.**

Agree.

> users can define any field they want even if it is a custom field **or not**

There are custom fields, jira fields that we support, and fields we do not support. Which one are these?

> it is difficult to validate against them.

What is `them` here? We can validate the `customFields` object easily against all the Jira fields we support.